clippy_lints/src/arc_new_in_vec_from_slice.rs

```diff
@@ -0,0 +1,84 @@
+use clippy_utils::diagnostics::span_lint_and_help;
+use clippy_utils::higher::VecArgs;
+use clippy_utils::last_path_segment;
+use clippy_utils::macros::{root_macro_call_first_node, MacroCall};
+use rustc_hir::{Expr, ExprKind, QPath, TyKind};
+use rustc_lint::{LateContext, LateLintPass};
+use rustc_session::{declare_lint_pass, declare_tool_lint};
+use rustc_span::sym;
+
+declare_clippy_lint! {
+    /// ### What it does
+    /// Checks for `Arc::new` in `vec![elem; len]`
+    ///
+    /// ### Why is this bad?
+    /// This will create `elem` once and clone it `len` times - doing so with `Arc`
+    /// is a bit misleading, as it will create references to the same pointer, rather
+    /// than different `Arc` instances.
+    ///
+    /// ### Example
+    /// ```rust
+    /// let v = vec![std::sync::Arc::new("some data".to_string()); 100];
+    /// ```
+    /// Use instead:
+    /// ```rust
+    /// let data = std::sync::Arc::new("some data".to_string());
+    /// let v = vec![data; 100];
+    /// ```
+    #[clippy::version = "1.62.0"]
+    pub ARC_NEW_IN_VEC_FROM_SLICE,
```

Comment:
```suggestion
    pub ARC_NEW_IN_VEC,
```

Forgot to suggest this, I'd omit the `from_slice` from the lint name

Reply:
But its ok to have `vec![]` if its not `from_slice` - are you sure omitting it from the name is ok?

Reply:
Maybe instead of `from_slice` it could be `from_elem`, since that makes it clear what it's checking for. The current name sounds like it's checking for `Vec::from(&[Arc])`.

Reply:
Ah, I see - will change to `from_elem`